src/configuration.js

```diff
@@ -297,15 +304,14 @@ export class Configuration {
 
     /**
      * Returns the global configuration instance. It will respect the environment variables.
-     * As opposed to this method, we can also get the SDK instance configuration via `sdk.config` property.
+     * When used inside SDK instance (`sdk.main()`), the return value is equal to `sdk.config`.
      *
      * @return {Configuration}
      */
     static getGlobalConfig() {
-        if (!Configuration.globalConfig) {
-            Configuration.globalConfig = new Configuration();
-        }
-
-        return Configuration.globalConfig;
+        return Configuration.storage.getStore() ?? (() => {
+            Configuration.globalConfig ??= new Configuration();
+            return Configuration.globalConfig;
+        })();
```

Comment:
Could this be written "normally"? 😄  I mean, it's kinda cool, but going from `??` into an IIFE is a bit hardcore.

Reply:
its not _that_ hardcode, but that IIFE made me smile too :D

Reply:
I paid for the whole JS grammar, I’m using the whole JS grammar!

But understood, I already hate it :) hope it is still legible enough at least

Reply:
i gave green light because to me it is readable enough, its actually quite smart way to do this on so few lines. but we can be more readable for sure:

```ts
if (Configuration.storage.getStore()) {
  return Configuration.storage.getStore();
}

Configuration.globalConfig ??= new Configuration();
return Configuration.globalConfig;
```